packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -291,6 +302,24 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
     assert(removed);
   }
 
+  void _updateScaffolds() {
+    for (final ScaffoldState scaffold in _scaffolds) {
+      if (_isRoot(scaffold)) {
+        scaffold._updateSnackBar();
+        scaffold._updateMaterialBanner();
```

Comment:
Should we check to see here if `_materialBanners.isNotEmpty` and `_snackBars.isNotEmpty` before updating each? `_updateScaffolds` could be called for more than one reason now.

Reply:
You may end up pushing a duplicate snackbar for example if it has already been pushed and then a banner update comes in.

Reply:
Actually, this may be better handled in the Scaffold, see comment below.